pkg/util/globalerror/grpc.go

```diff
@@ -7,12 +7,23 @@ import (
 
 	"github.com/gogo/status"
 	"github.com/grafana/dskit/grpcutil"
+	"google.golang.org/grpc"
 	"google.golang.org/grpc/codes"
 	grpcstatus "google.golang.org/grpc/status"
 
 	"github.com/grafana/mimir/pkg/mimirpb"
 )
 
+var grpcClientConnectionIsClosingErr string
+
+func init() {
+	// Ignore deprecation warning for now
+	//nolint:staticcheck
+	if stat, ok := grpcutil.ErrorToStatus(grpc.ErrClientConnClosing); ok && stat.Code() == codes.Canceled {
+		grpcClientConnectionIsClosingErr = stat.Message()
+	}
```

Comment:
What if the condition isn't met? The string would be empty. Should we panic then to make sure that the code doesn't compile?

Reply:
Hi @colega,
My idea in this case was to consider all errors with `codes.Canceled` as `context canceled`.

Reply:
Or better:
- if `grpc.ErrClientConnClosing` gets removed (it will happen at a poi t), this code will not compile anymore, which is good
- if `grpc.ErrClientConnaclosing` gets modified  and it is not a gRPC error and its status is not `codes.Canceled` (not likely to happen, why should a deprecated error change?),  we’ll not filter out it anymore
- if `grpc.ErrClientConnaclosing` gets modified but it is still a gRPC error with status `codes.Canceled` (not likely to happen, why should a deprecated error change?),  we’ll not filter out the new message

Reply:
But the variable would become a lie.

Reply:
I personally prefer to have an empty string, to change a name of the variable, or to hardcode `grpc: the client connection is closing` than to panic. 
But I am flexible, I'll do what you prefer. So what do we want to do? Panic?

Reply:
I liked the hardcoded string as it was clear and tested, this is extra logic, extra things that can fail. But I don't seem to be aligned with @jhalterman on this, so I'm going to approve it as is.

Reply:
@colega, @jhalterman 
I also prefer the hardcoded version.
Anyway, I am going to merge this version, but if we agree to use the hardcoded string, I’ll prepare another PR.
Please let me know you are OK with it @jhalterman